pkg/controller/controller_clusterservicebroker.go

```diff
@@ -785,3 +805,24 @@ func convertClusterServicePlanListToMap(list []v1beta1.ClusterServicePlan) map[s
 
 	return ret
 }
+
+func markAsServiceCatalogManagedResource(obj metav1.Object, broker *v1beta1.ClusterServiceBroker) {
+	if isServiceCatalogManagedResource(obj) {
+		return
+	}
+
+	var blockOwnerDeletion = false
+	controllerRef := *metav1.NewControllerRef(broker, v1beta1.SchemeGroupVersion.WithKind("ClusterServiceBroker"))
+	controllerRef.BlockOwnerDeletion = &blockOwnerDeletion
+
+	obj.SetOwnerReferences(append(obj.GetOwnerReferences(), controllerRef))
+}
+
+func isServiceCatalogManagedResource(resource metav1.Object) bool {
```

Comment:
I'm not sure about this one. My reservation is that `ClusterServiceClass` and `ClusterServicePlan` already have a field (`spec.ClusterServiceBrokerName`) that carries information about which `ClusterServiceBroker` manages the resource.

*does some research*

Alright, I think I understand this now. I hate to do design in this PR, another option to achieve the same behavior (ignoring user-defined serviceclasses/serviceplans) would be to have an annotation saying that the serviceclass/plan is user-defined. Have you considered that yet? I guess the merit of this approach over the opt-in annotation would be that a user doesn't have to know about the special annotation and will get the right behavior, in the naive path. WDYT?

Reply:
One thing I wonder about is whether setting the owner reference will have some unanticipated effects on garbage collection.

Reply:
I went down this path for a couple reasons:

* When a user makes a custom plan, `ClusterServiceBrokerName` will be populated in their custom plan too. That's not enough to tell them apart.
* Anything that wasn't made by our controller is automatically considered user defined. So that's a nicer UX.
* We already wanted to add controller references anyway (see https://github.com/kubernetes-incubator/service-catalog/issues/1432).
* As far as I understand owner / controller references, what we are trying to indicate "our controller made this, don't mess with it" fits perfectly into how they are supposed to be used.
* I am explicitly turning off the garbage collection flag, `BlockOwnerDeletion`  so that we don't affect current delete behavior.

Reply:
>Anything that wasn't made by our controller is automatically considered user defined. So that's a nicer UX.

I'm +1 on this approach because of this alone